libraries/chain/controller.cpp

```diff
@@ -723,6 +724,9 @@ struct controller_impl {
    ~controller_impl() {
       thread_pool.stop();
       pending.reset();
+      //using the presence of a row in datebase_header index to indicate controller had a successful startup()
```

Comment:
eh something better?

Reply:
Move the chain_plugin where you know if started successfully?

Reply:
I can't log the startup in chain_plugin because I need to to do the log before replay but after snapshot is loaded. So, it felt weird to do the startup log in controller but the shutdown log in chain_plugin.

I'm not sure doing it in chain_plugin really helps either since there are so many exceptions that can start unrolling the world at any time.

Actually I'm not really even sure this here works well enough: what if an exception is thrown half way through loading a snapshot. Then once everything unrolls the dtor is fired and if datebase_header made it then it does a snapshot with a half initialized state? That's what I want to avoid.

But I also wanted to avoid adding a `bool yep_controller_started_up`. It's kinda wild there isn't a clear indicator anywhere :thinking:

Reply:
In `chain_plugin::plugin_startup` you could add a `catch(...)` after the `database_guard_exception` catch and add a `my->chain.reset();`. Then in `chain_plugin::plugin_shutdown()` you could check `my->chain` to know if it is valid, if so then log the integrity.

Reply:
I capitulated and just added a bool :confused: